sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/EmptyNestedBlock.cs

```diff
@@ -0,0 +1,211 @@
+
+            public override void VisitForBlock(ForBlockSyntax node)
+            {
+                if (!node.Statements.Any() && NoCommentsBefore(node.NextStatement))
+                {
+                    emptyInnerBlocks.Add(node.ForStatement);
+                }
+            }
+
+            public override void VisitForEachBlock(ForEachBlockSyntax node)
+            {
+                if (!node.Statements.Any() && NoCommentsBefore(node.NextStatement))
+                {
+                    emptyInnerBlocks.Add(node.ForEachStatement);
+                }
+            }
+
+            public override void VisitMultiLineIfBlock(MultiLineIfBlockSyntax node)
+            {
+                if (!node.Statements.Any() && NoCommentsBefore(node.EndIfStatement))
+                {
+                    emptyInnerBlocks.Add(node.IfStatement);
+                }
+            }
+
+            public override void VisitSelectBlock(SelectBlockSyntax node)
+            {
+                if (!node.CaseBlocks.Any())
+                {
+                    emptyInnerBlocks.Add(node.SelectStatement);
+                }
+            }
+
+            public override void VisitTryBlock(TryBlockSyntax node)
+            {
+                if (!node.CatchBlocks.Any())
+                {
+                    VerifyTryBlock(node, node.FinallyBlock);
+                    VerifyFinallyBlock(node.FinallyBlock, node.EndTryStatement);
+                }
+                else if (node.FinallyBlock == null)
+                {
+                    VerifyTryAndMostCatches(node);
+                    VerifyCatchBlock(node.CatchBlocks[node.CatchBlocks.Count - 1], node.EndTryStatement);
+                }
+                else
+                {
+                    VerifyTryAndMostCatches(node);
+                    VerifyCatchBlock(node.CatchBlocks[node.CatchBlocks.Count - 1], node.FinallyBlock);
+                    VerifyFinallyBlock(node.FinallyBlock, node.EndTryStatement);
+                }
+            }
+
+            public override void VisitUsingBlock(UsingBlockSyntax node)
+            {
+                if (!node.Statements.Any() && NoCommentsBefore(node.EndUsingStatement))
+                {
+                    emptyInnerBlocks.Add(node.UsingStatement);
+                }
+            }
+
+            public override void VisitWhileBlock(WhileBlockSyntax node)
+            {
+                if (!node.Statements.Any() && NoCommentsBefore(node.EndWhileStatement))
+                {
+                    emptyInnerBlocks.Add(node.WhileStatement);
+                }
+            }
+
+            public override void VisitWithBlock(WithBlockSyntax node)
+            {
+                if (!node.Statements.Any() && NoCommentsBefore(node.EndWithStatement))
+                {
+                    emptyInnerBlocks.Add(node.WithStatement);
+                }
+            }
+
+            // Skip the last catch as that one will be followed by Finally or by End Try
```

Comment:
put this above the for loop, replace the comment with "verifies all catches except the last one"